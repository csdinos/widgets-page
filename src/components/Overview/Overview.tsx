import {Container, Button, Grid, Typography} from '@material-ui/core'
import {DeleteModal, WidgetListing} from '../'
import {useHistory} from 'react-router-dom'

export const Overview = () => {
  const history = useHistory()

  return (
    <Container maxWidth="xl">
      <Grid container
            direction="row"
            alignItems="center"
            justify="center"
            spacing={2}
      >
        <Grid item xs={8}>
          <Typography variant="h3" gutterBottom>
            Available widgets
          </Typography>
        </Grid>
        <WidgetListing />
        <Grid item xs={12} id="new-widget-btn">
          <Button
            variant="contained"
            color="primary"
            fullWidth
            onClick={() => history.push('/create/language')}
          >New</Button>
        </Grid>
      </Grid>
      <DeleteModal />
    </Container>
  )
}