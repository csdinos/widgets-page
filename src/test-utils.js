import React from 'react'
import {render as rtlRender} from '@testing-library/react'
import {createStore} from 'redux'
import {Provider} from 'react-redux'
import {reducer} from './reducers'

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({children}) {
    return <Provider store={store}>{children}</Provider>
  }

  return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
}

export const initialState = {
  widgets: [],
  newWidget: {
    id: -1,
    name: '',
    language: '',
  },
  deleteModalId: null,
  hasLoaded: false,
  shouldUpdateStorage: false
}

export * from '@testing-library/react'
export {render}