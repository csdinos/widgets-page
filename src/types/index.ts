export type Widget = {
  id: number;
  name: string;
  language: string;
}
export type State = {
  widgets: Array<Widget>
  newWidget: Widget
  deleteModalId: number | null,
  hasLoaded: boolean,
  shouldUpdateStorage: boolean
}