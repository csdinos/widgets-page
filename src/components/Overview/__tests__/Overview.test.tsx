import React from 'react'
import {initialState, render} from '../../../test-utils'
import {Overview} from '../Overview'
import {State} from '../../../types'

const state: State = {
  ...initialState,
  widgets: [{
    name: 'Doge',
    language: 'Meme',
    id: 1994
  }],
  hasLoaded: true
}

describe('<Overview />', () => {
  it('When we have an empty initial state, it should display an empty overview page', async () => {
    const {asFragment} = render(<Overview/>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('When we have a state with 1 widget, it should display an empty overview page', async () => {
    const {asFragment} = render(<Overview/>, {initialState: state})
    expect(asFragment()).toMatchSnapshot()
  })
})