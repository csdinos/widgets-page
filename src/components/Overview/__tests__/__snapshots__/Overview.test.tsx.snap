// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overview /> When we have a state with 1 widget, it should display an empty overview page 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Available widgets
      </h3>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        class="MuiPaper-root MuiCard-root widget-card MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Name: Doge
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Language: Meme
          </p>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained delete-btn MuiButton-containedSecondary MuiButton-fullWidth"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Delete
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      id="new-widget-btn"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          New
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Overview /> When we have an empty initial state, it should display an empty overview page 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Available widgets
      </h3>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      id="new-widget-btn"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          New
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
